addons/controllers/webhooks_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	cert2 "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"knative.dev/pkg/webhook/certificates/resources"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/controllers/webhooks"
+)
+
+func TestWebhooks(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Webhooks TLS",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var (
+	setupLog = ctrl.Log.WithName("controllers").WithName("Addon")
+)
+var _ = Describe("when TLS secret is created", func() {
+
+	Context("If the certificates do not exists", func() {
+		It("should install new certificates", func() {
+			testTLSSecret := webhooks.WebhookTLS{
+				Ctx:           ctx,
+				K8sConfig:     k8sConfig,
+				CertPath:      certPath,
+				KeyPath:       keyPath,
+				Name:          webhookScrtName,
+				ServiceName:   webhookServiceName,
+				LabelSelector: "webhook-cert=self-managed",
+				Logger:        setupLog,
+				TLSSecret:     nil,
+				Namespace:     addonNamespace,
+				InvalidWithIn: time.Hour * 24,
+			}
+			err := testTLSSecret.Manage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testTLSSecret.TLSSecret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM).To(Equal(testTLSSecret.TLSSecret.Data[resources.ServerCert]))
+			key, err := keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err := keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+		})
+	})
+	Context("If the certificates does exists", func() {
```

Comment:
nit: typo `do exist`

Reply:
done

Reply:
Removed test as it is redundant. Other tests cover these same steps.